source/base/index_set.cc

```diff
@@ -232,7 +232,27 @@ IndexSet::get_view(const size_type begin, const size_type end) const
   return result;
 }
 
-
+std::vector<IndexSet>
+IndexSet::split_by_block(
+  const std::vector<types::global_dof_index> &dofs_per_block,
+  const IndexSet &                            whole_set)
+{
+  std::vector<IndexSet> partitioned;
+  const unsigned int    n_blocks = dofs_per_block.size();
+  partitioned.clear();
+
+  partitioned.resize(n_blocks);
+  types::global_dof_index start = 0;
+  types::global_dof_index sum   = 0;
+  for (unsigned int i = 0; i < n_blocks; ++i)
+    {
+      partitioned[i] = whole_set.get_view(start, start + dofs_per_block[i]);
+      start += dofs_per_block[i];
+      sum += partitioned[i].size();
+    }
+  AssertDimension(whole_set.size(), sum);
```

Comment:
For me, swapping them makes more sense, i.e. having the object that has to fulfill some condition on the left and the condition on the right.
```suggestion
  AssertDimension(sum, whole_set.size());
```

Reply:
okay, I agree